ciao-controller/internal/datastore/sqlite3db_test.go

```diff
@@ -1221,13 +1221,61 @@ func TestSQLiteDBUpdateTenant(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	if tenant.Name != "name2" || tenant.SubnetBits != 20 {
+	if tenant.Name != "name2" || tenant.SubnetBits != 20 || tenant.Permissions.PrivilegedContainers != true {
 		t.Fatal("update not successful")
 	}
 
 	db.disconnect()
 }
 
+func TestSQLiteDBTenantPermissions(t *testing.T) {
+	db, err := getPersistentStore()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tenantID := uuid.Generate().String()
+	config := types.TenantConfig{
+		Name:       "name1",
+		SubnetBits: 24,
+	}
+	config.Permissions.PrivilegedContainers = true
+
+	err = db.addTenant(tenantID, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tenant, err := db.getTenant(tenantID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !tenant.Permissions.PrivilegedContainers {
+		t.Fatal("Expected tenant permission set correctly")
+	}
+
+	ts, err := db.getTenants()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(ts) != 1 {
+		t.Fatalf("Unexpected number of tenants")
+	}
+
+	if !ts[0].Permissions.PrivilegedContainers {
+		t.Fatal("Expected tenant permission set correctly")
```

Comment:
Does it matter if we don't delete the tenant or disconnect from the db on failure?

Reply:
No not really. I spotted this too. Each test is run in it's own temporary in-memory DB, everything is thrown out on test exit.